src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -684,15 +684,15 @@ abstract class ClassfileParser(reader: ReusableInstance[ReusableDataReader]) {
                 }
                 accept('>')
                 assert(xs.length > 0, tp)
-                debuglogResult("new existential")(newExistentialType(existentials.toList, typeRef(pre, classSym, xs.toList)))
+                debuglogResult("new existential")(newExistentialType(existentials.toList, typeRef(pre, tsym, xs.toList)))
```

Comment:
maybe the intervening 30 LOC could be moved to a method, then these refs are closer to the pattern they come from